osquery/tables/system/windows/windows_search.cpp

```diff
@@ -0,0 +1,244 @@
+// osquery/tables/system/windows/windows_search.cpp
+
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <atldbcli.h>
+#include <codecvt>
+#include <comutil.h>
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/windows/strings.h>
+#include <osquery/utils/conversions/windows/windows_time.h>
+#include <sstream>
+#include <strsafe.h>
+#include <vector>
+#include <windows.h>
+
+#pragma comment(lib, "comsuppw.lib")
+
+namespace osquery {
+namespace tables {
+
+LONGLONG dateToUnixTime(const DATE date) {
+  SYSTEMTIME st;
+  FILETIME ft;
+
+  VariantTimeToSystemTime(date, &st);
+  SystemTimeToFileTime(&st, &ft);
+  return filetimeToUnixtime(ft);
+}
+
+// This helper function can print some propvariants and handles BSTR vectors
+void writePropVariant(REFPROPVARIANT variant, std::wstringstream& wss) {
+  if (variant.vt == (VT_ARRAY | VT_BSTR) && variant.parray->cDims == 1) {
+    BSTR* pBStr;
+    HRESULT hr =
+        SafeArrayAccessData(variant.parray, reinterpret_cast<void**>(&pBStr));
+    if (SUCCEEDED(hr)) {
```

Comment:
indentation can be improved here by checking by if (FAILED(hr)) instead and failing fast